app/src/main/java/com/valhalla/thor/data/launcher/FreezerShortcutManager.kt

```
@@ -287,36 +287,72 @@ class FreezerShortcutManager(
+     * while per-app pins are gated to user apps, which is exactly what makes it worth fixing before
+     * that gate moves rather than after.
+     *
+     * Handing the result to both consumers is not a micro-optimisation. `getApplicationIcon(String)`
+     * resolves through `getApplicationInfo(pkg, sDefaultFlags)`, and `sDefaultFlags` is
+     * GET_SHARED_LIBRARY_FILES — no match flags at all — so widening the label's lookup on its own
+     * would un-mask the icon's and trade "the icon never refreshes" for "the icon is a blank tile".
+     * The `ApplicationInfo` overloads take flags out of the question for both.
+     */
+    private fun applicationInfo(packageName: String): ApplicationInfo? = try {
+        context.packageManager.getApplicationInfo(packageName, AppFreezeStateReader.MATCH_FLAGS)
     } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)